<launch>

  <arg name="uav_name" default="$(optenv UAV_NAME uav)"/>

  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix" value="debug_roslaunch"/>

  <arg name="ignore_mask" default="$(optenv IGNORE_MASK false)" />
  <arg unless="$(arg ignore_mask)" name="mask_filename" value="$(find uav_detect)/masks/nerian_eagleone_mask.bmp"/>
  <arg     if="$(arg ignore_mask)" name="mask_filename" value=""/>

  <arg name="standalone" default="true" />
  <arg name="manager" default="detection_nodelet_manager" />
  <arg name="det_nodelet" value="uav_detect/PCLDetector" />
  <arg unless="$(arg standalone)" name="nodelet_load" value="load $(arg det_nodelet) $(arg manager)"/>
  <arg     if="$(arg standalone)" name="nodelet_load" value="standalone $(arg det_nodelet)"/>

  <group ns="$(arg uav_name)">

    <node pkg="nodelet" type="nodelet" name="uav_detection" args="$(arg nodelet_load)" output="screen" launch-prefix="$(arg launch_prefix)">
      
      <param name="uav_name" value="$(arg uav_name)" />
      <param name="path_to_mask" value="$(arg mask_filename)" />
      <rosparam file="$(find uav_detect)/config/detection_params_sim.yaml" />
    
      <remap from="~pc" to="/os1_cloud_node/points" />

    </node>
  
  </group>

</launch>
